Packs/ZeroFox/Integrations/ZeroFox/ZeroFox.py

```diff
@@ -112,652 +711,840 @@ def remove_none_dict(input_dict: Dict) -> Dict:
+    credentials_response = client.get_cti_compromised_credentials(email)
+    botnet_credentials_response = client\
+        .get_cti_botnet_compromised_credentials(email)
+
+    outputs = get_compromised_email_content(
+        email_addresses_response,
+        credentials_response,
+        botnet_credentials_response,
+    )
+
+    if len(outputs) == 0:
+        return CommandResults(
+            outputs=outputs,
+            readable_output="No compromised emails were found",
+            outputs_prefix="ZeroFox.CompromisedEmails",
+        )
+    return CommandResults(
+        outputs=outputs,
+        readable_output=tableToMarkdown(
+            "Compromised email Summary",
+            outputs,
+        ),
+        outputs_prefix="ZeroFox.CompromisedEmails",
+    )
+
+
+def malicious_ip_command(client: ZFClient, args: Dict):
+    ip: str = args.get("ip", "")
+
+    botnet_response = client.get_cti_botnet(ip)
+    phishing_response = client.get_cti_phishing(ip=ip)
+
+    outputs = get_malicious_ip_content(
+        botnet_response,
+        phishing_response,
+    )
+
+    if len(outputs) == 0:
+        return CommandResults(
+            readable_output="No malicious ips were found",
+            outputs=outputs,
+            outputs_prefix="ZeroFox.MaliciousIPs",
+        )
+    return CommandResults(
+        outputs=outputs,
+        readable_output=tableToMarkdown("Malicious ip Summary", outputs),
+        outputs_prefix="ZeroFox.MaliciousIPs",
+    )
+
 
+def malicious_hash_command(client: ZFClient, args: Dict):
+    hash: str = args.get("hash", "")
 
-''' COMMANDS MANAGER / SWITCH PANEL '''
+    outputs = []
+    for hash_type in ["md5", "sha1", "sha256", "sha512"]:
+        hash_type_response = client.get_cti_malware(hash_type, hash)
+        outputs += get_malicious_hash_content(hash_type, hash_type_response)
+
+    if len(outputs) == 0:
+        return CommandResults(
+            readable_output="No malicious hashes were found",
+            outputs=outputs,
+            outputs_prefix="ZeroFox.MaliciousHashes",
+        )
+    return CommandResults(
+        outputs=outputs,
+        readable_output=tableToMarkdown("Malicious hash Summary", outputs),
+        outputs_prefix="ZeroFox.MaliciousHashes",
+    )
+
+
+def search_exploits_command(client: ZFClient, args: Dict):
+    since: str = args.get("since", "")
+
+    exploits_response = client.get_cti_exploits(since)
+
+    outputs = get_exploits_content(exploits_response)
+
+    if len(outputs) == 0:
+        return CommandResults(
+            outputs=outputs,
+            readable_output="No exploits were found",
+            outputs_prefix="ZeroFox.Exploits",
+        )
+    return CommandResults(
+        outputs=outputs,
+        readable_output=tableToMarkdown("Exploit Search Summary", outputs),
+        outputs_prefix="ZeroFox.Exploits",
+    )
+
+
+""" COMMANDS MANAGER / SWITCH PANEL """
 
 
 def main():
```

Comment:
Instead of using 'demisto.params()', declare a variable, 'params = demisto.params()', and use it.